src/common/snippets/src/op/memory_access.cpp

```diff
@@ -12,32 +12,66 @@ namespace ngraph {
 namespace snippets {
 namespace op {
 
-MemoryAccess::MemoryAccess(const Output<Node>& x, const size_t count, const size_t offset) : Op({x}), m_count(count), m_offset(offset) {}
+MemoryAccess::MemoryAccess(const OutputVector& arguments) : Op(arguments) {}
 
 bool MemoryAccess::visit_attributes(AttributeVisitor& visitor) {
-    visitor.on_attribute("count", m_count);
-    visitor.on_attribute("offset", m_offset);
+    for (size_t i = 0; i < m_input_ports.size(); ++i) {
+        auto port = m_input_ports[i];
+        visitor.on_attribute("count_in_" + std::to_string(i), port.m_count);
+        visitor.on_attribute("offset_in_" + std::to_string(i), port.m_offset);
+    }
+    for (size_t i = 0; i < m_output_ports.size(); ++i) {
+        auto port = m_output_ports[i];
+        visitor.on_attribute("count_out_" + std::to_string(i), port.m_count);
+        visitor.on_attribute("offset_out_" + std::to_string(i), port.m_offset);
+    }
     return true;
 }
 
-size_t MemoryAccess::get_count() const {
-    return m_count;
+void MemoryAccess::set_input_port_descriptor(const PortDescriptor& desc, const size_t i) {
+    // Logic is as same as ov::Node::get_input_descriptor
+    while (m_input_ports.size() <= i) {
```

Comment:
Why don't we check that `i < get_input_size()`? It's unclear why we can set 100 input descriptors for an operation that has only 2 inputs. Same for outputs.